### ISSUE TITLE
Keras Conv3DTranspose inconsistency between padding, output_shape and output_padding

### ISSUE THREAD

**Issue by bhushan23** [tensorflow/tensorflow#39469]
## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv3DTranspose

## Description of issue (what needs changing):

Output shape computation is shown as below on above documentation
```
new_depth = ((depth - 1) * strides[0] + kernel_size[0] - 2 * padding[0] +
output_padding[0])
new_rows = ((rows - 1) * strides[1] + kernel_size[1] - 2 * padding[1] +
output_padding[1])
new_cols = ((cols - 1) * strides[2] + kernel_size[2] - 2 * padding[2] +
output_padding[2])
```
but padding is either 'valid' or 'same'
so, is padding computed based on traditional convolution computation (ref: https://www.tensorflow.org/api_docs/python/tf/nn/convolution) and then used here?
This is unclear from current documentation how `same`/`valid` mode is being used.

### Clear description

Clarification about how these modes are reflected in computing actual padding and then used in specified formula.

### Computation of output shape
`deconv_output_length` from keras/utils/conv_utils.py is used for computing the output shape considering output_padding which should be reflected into documentation concisely
ref: https://github.com/tensorflow/tensorflow/blob/dd2ea875d92eeb83e81b1cb92e29e61d488e98b2/tensorflow/python/keras/utils/conv_utils.py#L168

**Comment by tensorflowbutler** [comment]
Hi There,

 This is a stale issue. As you are using an older version of tensorflow, we are checking to see if you still need help on this issue. Please test the issue with the latest TensorFlow (TF2.7 and tf-nightly). If the issue still persists with the newer versions of TF, please feel free to open it in [keras-team/keras](https://github.com/keras-team/keras/issues) repository by providing details about the issue and a standalone code to reproduce the issue. Thanks! 

 Please note that Keras development has moved to a separate Keras-team/keras repository to focus entirely on only Keras. Thanks!
